### ISSUE TITLE
`alternating_update` breaks for certain vertextypes

### ISSUE THREAD

**Issue by b-kloss** [mtfishman/ITensorNetworks.jl#140]
Algorithms based on alternating update should work for any vertextype, given that the input state and operator are TTNs. Setting up a `path_graph` with non-standard `vertextype` as in the following test

```
using ITensors
using ITensorNetworks
using ITensorNetworks: exponentiate_updater,add_edge!
using Test

@testset "MPS TDVP with other vertexnames" begin
    ## make path_graph with vertextype(g)==Tuple(Int,Int,Vararg{Int})
    g = ITensorNetworks.NamedGraph([(1,1),(1,2,1),(1,2,3,4),(1,4)])
    add_edge!(g,(1,1)=>(1,2,1))
    add_edge!(g,(1,2,1)=>(1,2,3,4))
    add_edge!(g,(1,2,3,4)=>(1,4))
    @show ITensorNetworks.vertextype(g)
    
    s = siteinds("S=1/2", g)
    os=ITensorNetworks.heisenberg(s)
    H = TTN(os, s)
    ψ0 = random_ttn(s)
    @test is_tree(H)
    @test is_tree(ψ0)
    
    # Time evolve forward:
    @test (ψ1 = tdvp(H, -0.1im, ψ0; nsteps=1, cutoff=1e-10, nsites=1))
end
```
leads to output
```
julia> include("test_treetensornetworks/test_solvers/test_tdvp_anyvertextype.jl")
ITensorNetworks.vertextype(g) = Tuple{Int64, Int64, Vararg{Int64}}
MPS TDVP with other vertexnames: Error During Test at /mnt/home/bkloss/.julia/v1.9/dev/ITensorNetworks/test/test_treetensornetworks/test_solvers/test_tdvp_anyvertextype.jl:25
  Test threw exception
  Expression: ψ1 = tdvp(H, -0.1im, ψ0; nsteps = 1, cutoff = 1.0e-10, nsites = 1)
  AssertionError: is_tree(itensor_network)
  Stacktrace:
    [1] TreeTensorNetwork (repeats 2 times)
      @ ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/ttn.jl:16 [inlined]
    [2] #orthogonalize#537
      @ ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/abstracttreetensornetwork.jl:108 [inlined]
    [3] orthogonalize(tn::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}, edge::NamedGraphs.NamedEdge{Tuple{Int64, Int64, Vararg{Int64}}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/abstracttreetensornetwork.jl:107
    [4] orthogonalize(ψ::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}, root_vertex::Tuple{Int64, Int64}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/abstracttreetensornetwork.jl:100
    [5] orthogonalize(ψ::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}, root_vertex::Tuple{Int64, Int64})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/abstracttreetensornetwork.jl:92
    [6] region_update(updater::typeof(exponentiate_updater), projected_operator::ProjTTN{Tuple{Int64, Int64, Vararg{Int64}}}, state::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}; normalize::Bool, outputlevel::Int64, which_sweep::Int64, sweep_plan::Vector{Tuple{Any, NamedTuple{(:substep, :time_step), Tuple{Int64, ComplexF64}}}}, which_region_update::Int64, region_kwargs::NamedTuple{(:substep, :time_step, :maxdim, :mindim, :cutoff, :noise), Tuple{Int64, ComplexF64, Int64, Int64, Float64, Float64}}, region_observer!::DataFrames.DataFrame, updater_kwargs::NamedTuple{(), Tuple{}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/update_step.jl:188
    [7] sweep_update(solver::typeof(exponentiate_updater), projected_operator::ProjTTN{Tuple{Int64, Int64, Vararg{Int64}}}, state::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}; normalize::Bool, outputlevel::Int64, region_update_printer::Function, region_observer!::DataFrames.DataFrame, which_sweep::Int64, sweep_params::NamedTuple{(:maxdim, :mindim, :cutoff, :noise), Tuple{Int64, Int64, Float64, Float64}}, sweep_plan::Vector{Any}, updater_kwargs::NamedTuple{(), Tuple{}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/update_step.jl:76
    [8] macro expansion
      @ ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/alternating_update.jl:75 [inlined]
    [9] macro expansion
      @ ./timing.jl:393 [inlined]
   [10] alternating_update(updater::typeof(exponentiate_updater), projected_operator::ProjTTN{Tuple{Int64, Int64, Vararg{Int64}}}, init_state::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}; checkdone::ITensorNetworks.var"#697#701"{ITensorNetworks.var"#697#698#702"}, outputlevel::Int64, nsweeps::Int64, sweep_observer!::DataFrames.DataFrame, sweep_printer::Function, write_when_maxdim_exceeds::Nothing, updater_kwargs::NamedTuple{(), Tuple{}}, kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:sweep_plan, :cutoff), Tuple{Vector{Any}, Float64}}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/alternating_update.jl:74
   [11] alternating_update(updater::Function, H::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}, init_state::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}; kwargs::Base.Pairs{Symbol, Any, NTuple{5, Symbol}, NamedTuple{(:nsweeps, :sweep_observer!, :sweep_plan, :updater_kwargs, :cutoff), Tuple{Int64, DataFrames.DataFrame, Vector{Any}, NamedTuple{(), Tuple{}}, Float64}}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/alternating_update.jl:102
   [12] alternating_update
      @ ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/alternating_update.jl:95 [inlined]
   [13] tdvp(updater::typeof(exponentiate_updater), operator::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}, t::ComplexF64, init_state::TreeTensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}; time_step::ComplexF64, nsites::Int64, nsteps::Int64, order::Int64, sweep_observer!::DataFrames.DataFrame, root_vertex::Tuple{Int64, Int64}, reverse_step::Bool, updater_kwargs::NamedTuple{(), Tuple{}}, kwargs::Base.Pairs{Symbol, Float64, Tuple{Symbol}, NamedTuple{(:cutoff,), Tuple{Float64}}})
      @ ITensorNetworks ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/tdvp.jl:92
   [14] #tdvp#710
      @ ~/.julia/v1.9/dev/ITensorNetworks/src/treetensornetworks/solvers/tdvp.jl:130 [inlined]
   [15] macro expansion
      @ /mnt/sw/nix/store/fms6zspq4gnq3x8bps8i9wx4lhd6j8s3-julia-1.9.0/share/julia/stdlib/v1.9/Test/src/Test.jl:478 [inlined]
   [16] macro expansion
      @ ~/.julia/v1.9/dev/ITensorNetworks/test/test_treetensornetworks/test_solvers/test_tdvp_anyvertextype.jl:25 [inlined]
   [17] macro expansion
      @ /mnt/sw/nix/store/fms6zspq4gnq3x8bps8i9wx4lhd6j8s3-julia-1.9.0/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
   [18] top-level scope
      @ ~/.julia/v1.9/dev/ITensorNetworks/test/test_treetensornetworks/test_solvers/test_tdvp_anyvertextype.jl:8
Test Summary:                   | Pass  Error  Total  Time
MPS TDVP with other vertexnames |    2      1      3  0.1s
ERROR: LoadError: Some tests did not pass: 2 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /mnt/home/bkloss/.julia/v1.9/dev/ITensorNetworks/test/test_treetensornetworks/test_solvers/test_tdvp_anyvertextype.jl:6
```

**Comment by b-kloss** [comment]
Digging a bit into this, it seems that the network gets disconnected when the first orthogonalization on an edge is called. 
```
ITensorNetworks.vertextype(g) = Tuple{Int64, Int64, Vararg{Int64}}
post_order_dfs_edges(ψ0, (1, 4)) = NamedGraphs.NamedEdge{Tuple{Int64, Int64, Vararg{Int64}}}[(1, 1) => (1, 2, 1), (1, 2, 1) => (1, 2, 3, 4), (1, 2, 3, 4) => (1, 4)]
region = [(1, 4)]
```
the above is output before entering `orthogonalize`, and what follows is from adding `@show` in the appropriate places in `orthogonalize` ( the functions defined in lines 92 and 107)
```
e = (1, 1) => (1, 2, 1)
typeof(atn) = ITensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}}
atn = ITensorNetwork{Tuple{Int64, Int64, Vararg{Int64}}} with 4 vertices:
4-element Vector{Tuple{Int64, Int64, Vararg{Int64}}}:
 (1, 1)
 (1, 2, 1)
 (1, 2, 3, 4)
 (1, 4)

and 2 edge(s):
(1, 1) => (1, 2, 1)
(1, 2, 3, 4) => (1, 4)

with vertex data:
4-element Dictionaries.Dictionary{Tuple{Int64, Int64, Vararg{Int64}}, Any}
       (1, 1) │ ((dim=2|id=901|"S=1/2,Site,n=1×1"), (dim=1|id=148))
    (1, 2, 1) │ ((dim=2|id=807|"S=1/2,Site,n=1×2×1"), (dim=1|id=148))
 (1, 2, 3, 4) │ ((dim=2|id=737|"S=1/2,Site,n=1×2×3×4"),)
       (1, 4) │ ((dim=2|id=861|"S=1/2,Site,n=1×4"),)
```

**Comment by mtfishman** [comment]
Thanks for the report, should be easy to fix.

**Comment by b-kloss** [comment]
It seems like it was a mistake on my side, caused by calling random_ttn without `link_space` kwarg. I'll update once I know more, but maybe we should set a default of 1 in random_ttn to avoid this behaviour in the future?

**Comment by b-kloss** [comment]
It's indeed fixed by setting `link_space` in the call to `random_ttn`. This behaviour could be fixed by setting default in the definition of `random_ttn`.

**Comment by mtfishman** [comment]
Ok that explains it. Yeah, it probably make sense to set the default `link_space` to `1`.

**Comment by mtfishman** [comment]
Closed by #141.
